CorsixTH/Lua/epidemic.lua

```diff
@@ -304,11 +304,10 @@ function Epidemic:markForVaccination(patient)
     patient.marked_for_vaccination = true
     patient:setMood("epidemy4","deactivate")
     patient:setMood("epidemy2","activate")
+    self.world.ui:playSound("vaccin.wav")
```

Comment:
This can't be good.

Only the local player hospital may use the UI, while all hospitals may have epidemics.

Reply:
It's a function triggered by UI, the timer icon.

Reply:
It's still looks like mixing internal administrative changes with performing UI notifications to me.

Life gets much easier if you follow one way to do things for all occurrences. With this solution we get some UI going through the hospital interface and other UI actions taking shortcuts through other paths. If we ever want to change UI notifications, or wamnt to have epidemcis handling by non-visible hospitals, you'll have to check everywhere rather than just modify the hospital interface functions.

Unfortunately it not possible to enforce a policy.

Reply:
I don't think having the sound play here is out of scope of other code.
Earthquakes in world request sound to be played from inside that code too.

Reply:
Earth quakes are a bit different because they affect the world rather than a single hospital.

I agree that a function for playing a vaccine sound is a bit overkill in the amount of code, although I would still do it because it preserves the idea that UI always runs through the hospital.   
Coding is difficult. By making things predictable in doing the same thing in the same way every time we waste less time on handling complicated edge cases that don't fit in the general pattern. Code in general eventually becomes easier to grasp (eg by following the above,  if you want to know or change sounds of a hospital, you *know* that looking in the hospital is enough).

In this case, having a dedicated function tor this sound would indeed work. Other options are writing a general `Hospital:playSound`  function (likely more sounds can be rerouted in that way), or alternatively, ask the hospital for permission. with eg `Hospital:isSoundEnabled()` although that would be a slightly different way of doing things  where one may need to consider if all the currently implemented rerouting should be changed as well (to preserve the "do the same thing in the same way all the time").